server/middleware/security/authentication.js

```diff
@@ -224,6 +212,66 @@ module.exports = function (app) {
           this.authenticateAppByApiKey(req, res, next);
         });
       };
+    },
+
+    authenticateAppByEncryptedApiKey: (req, res, next) => {
+      required('api_key_enc')(req, res, (e) => {
+        if (e) return next(e);
+        const apiKeyEnc = req.required.api_key_enc;
+        try {
+          const apiKey = jwt.verify(apiKeyEnc, secret).apiKey;
+          findApplicationByKey(apiKey, req, next);
+        } catch (e) {
+          return res.send(400, e.message);
+        }
+      });
+    },
+
+    authenticateService: (req, res, next) => {
+      const apiKey = req.required.api_key;
+      const apiKeyEnc = jwt.sign({apiKey}, secret, { expiresIn: '1min' });
+      const service = req.params.service;
+      passport.authenticate(service, {
+        callbackURL: `${config.host.api}/connected-accounts/${service}/callback?api_key_enc=${apiKeyEnc}`,
+        // TODO set proper scope
```

Comment:
Might depend what other data you needed to access through the API during the onboarding? As far as the auth flow goes, this is enough. Can be tweaked in separate PR.